integration/jnifuse/fs/src/main/java/alluxio/jnifuse/struct/FuseFileInfo.java

```diff
@@ -11,24 +11,38 @@
 
 package alluxio.jnifuse.struct;
 
+import alluxio.jnifuse.utils.NativeLibraryLoader;
 import jnr.ffi.Runtime;
+import jnr.ffi.Struct;
 
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
 
-public class FuseFileInfo extends ru.serce.jnrfuse.struct.FuseFileInfo {
-  private final ByteBuffer buffer;
+public class FuseFileInfo extends Struct {
+
+  public u_int64_t fh;
+
+  public Signed32 flags;
+
+  public ByteBuffer buffer;
 
   public FuseFileInfo(Runtime runtime, ByteBuffer buffer) {
     super(runtime);
     this.buffer = buffer;
-    // depends on the arch
     this.buffer.order(ByteOrder.LITTLE_ENDIAN);
   }
 
   public static FuseFileInfo of(ByteBuffer buffer) {
     Runtime runtime = Runtime.getSystemRuntime();
-    FuseFileInfo fi = new FuseFileInfo(runtime, buffer);
+    // select the actual FuseFileInfo by loaded version
+    NativeLibraryLoader.LoadState state = NativeLibraryLoader.getLoadState();
+    if (state == NativeLibraryLoader.LoadState.NOT_LOADED) {
+      // TODO which exception to throw?
+      throw new RuntimeException("NativeLibraryLoader is not loaded");
```

Comment:
Can we give more detailed information in the exception message?
like libjnifuse library is not loaded

Reply:
If we are throwing RuntimeException, this `of` function is called in `open/read/write....Callback` functions which are called by libfuse threads.
libfuse threads cannot handle any java runtime exception and likely to crash.

Is the NOT_LOADED load state possible in this function?
I think it's not possible. Do we need this check and throw?

Reply:
Oh it's not possible in real cluster running, but it's possible in our testing....
then can you add descriptions to mention that this block is not possible in real cluster running and is added for tests purposes?
and add logs here.
Throw a checked exception instead of runtime exception.
@yyongycy what's your suggestions

Reply:
Comment added. 

This exception is not possible in production. Caller should not handle this situation.